apisix/plugin.lua

```diff
@@ -746,6 +746,10 @@ function _M.init_worker()
     -- see https://github.com/apache/apisix/issues/3286
     _M.load()
 
+    if is_http then
+        require("apisix.plugins.server-info").init_worker()
```

Comment:
`require("apisix.plugins.xxx")` will make the module import complex. We should use a separate file to hold the functons.

Reply:
Do you mean moving `server-info.lua` to elsewhere? But we should keep compatibility so that users could still use this plugin like before but with warning logs.
If only for `init_worker()` function, then because it's tightly coupled with symbols in `server-info.lua`, so it seems no clean way to separate it.
And I think it's not so necessary. `apisix/plugin.lua` is exactly used to load a specific plugin. In fact, in the same function, it also requires Prometheus plugin with a similar path:
https://github.com/apache/apisix/blob/09cc4946bc051452960975b4dd7436a2497df346/apisix/plugin.lua#L740

Reply:
Yes, but we don't require Prometheus plugin. The required module is "apisix.plugins.prometheus.exporter", not the Prometheus plugin itself. We need to do the same with server-info.

Reply:
@spacewander Split server-info.lua into server-info/common.lua and server-info/init_worker.lua, have a check please.